### DISCUSSION TITLE
eject usb drive

### DISCUSSION THREAD

**Discussion by ImSreyas** [sxyazi/yazi#1261] [1. Q&A]
Has anyone written any functionality for ejecting a USB drive?
I wish there was a way to eject the USB drive right from its mount point.

**Comment by matejdro** [comment]
I found that bashmount is a pretty good alternative: https://github.com/jamielinux/bashmount

**Reply by ImSreyas** [reply]
```
[manager]
prepend_keymap = [
	{ on = [ "e" ], run = ''' shell --confirm 'code "$@"' ''', desc = "Open the current dir in vscode" },
	{ on = [ "z" ], run = "plugin fzf", desc = "Jump to a directory, or reveal a file using fzf" },
	{ on = [ "Z" ], run = "plugin zoxide", desc = "Jump to a directory using zoxide" },
	{ on  = [ "y" ], run = [ "yank", '''
		shell --confirm 'echo "$@" | xclip -i -selection clipboard -t text/uri-list'
	''' ] },
	{ on = [ "E" ], run = ''' shell --block --confirm 'sudo eject $(findmnt -n -o source "$@")' ''', desc = "Eject the usb drive from mount point" },
]
```

Currently i have mapped 'E' on the mount point to eject the usb drive from the system...
and it works.... :)

**Reply by ImSreyas** [reply]
Thanks for recommending bashmount, :heart: :smile:  
it works well, so gonna use it for more advanced media management...

**Comment by sxyazi** [comment]
Made a plugin for this, https://github.com/yazi-rs/plugins/tree/main/mount.yazi
